src/main.c

```diff
@@ -351,18 +362,120 @@ key_binding_func_key (MBWindowManager   *wm,
   hd_dbus_send_event (s);
 }
 
+/* Close all input devices previously opened */
+static void
+close_input_devices (Display *dpy)
```

Comment:
The call to this function is missing on hildon-desktop termination.

Reply:
Yes, but there is no graceful termination of hildon-desktop I was able to find.

Reply:
What about lines coming after call to `gtk_main()` in the main function of main.c file?

https://github.com/maemo-leste/hildon-desktop/blob/95795a63ae6585d68a7e203dd407dfa97352d446/src/main.c#L814

Reply:
ok, I'll put a call to close_input_devices() there, but I doubt it will ever be reached.